sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqMessage.java

```diff
@@ -134,9 +135,10 @@ public RabbitMqMessage(byte[] body) {
     clusterId = null;
   }
 
-  public RabbitMqMessage(String routingKey, GetResponse delivery) {
-    this.routingKey = routingKey;
+  public RabbitMqMessage(GetResponse delivery) {
     delivery = serializableDeliveryOf(delivery);
+    this.routingKey =
```

Comment:
Found what I think is a bug. The routing key should *not* come from configuration, it should come from the delivered message. A queue can be bound to multiple routing keys; it isn't a given which was used.

Reply:
Yeah, although I think it wouldn't ultimately have any impact; I believe on publish this value is overwritten regardless. But it's still dumb.